clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -865,7 +881,8 @@ protected void close(Timer timer) {
             client.pollNoWakeup();
         }
 
-        resetGeneration();
+        resetGeneration("consumer proactively leaving group", false);
```

Comment:
Hmm.. We have already send the LeaveGroup when we invoke this. I guess we cannot actually guarantee that the consumer is still in the group when `onPartitionsRevoked` is called. Perhaps we need to move this up?

Reply:
Yes that's exactly what I meant above.